tests/fakeapi/aiohttp_hello.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+import asyncio
+
+import flask
+from flask import jsonify, redirect
+
+import connexion
+from aiohttp.web import Response as AioHttpResponse
+from connexion import NoContent, ProblemException, problem
+from connexion.lifecycle import ConnexionResponse
+
+
+@asyncio.coroutine
+def get_bye(name):
+    return AioHttpResponse(
+        text='Goodbye {}'.format(name)
+    )
+
+
+@asyncio.coroutine
+def aiohttp_str_response():
+    return ConnexionResponse(
+        body='str response'
+    )
+
+
+@asyncio.coroutine
+def aiohttp_non_str_non_json_response():
+    return ConnexionResponse(
+        body=1234
+    )
+
+
+@asyncio.coroutine
+def aiohttp_bytes_response():
+    return ConnexionResponse(
+        body=b'bytes response'
+    )
+
+
+@asyncio.coroutine
+def aiohttp_validate_responses():
+    return ConnexionResponse(
+        body=b'{"validate": true}'
+    )
+
+
+@asyncio.coroutine
+def aiohttp_post_greeting(name, **kwargs):
+    data = {'greeting': 'Hello {name}'.format(name=name)}
+    return ConnexionResponse(
+        body=data
+    )
+
+
+@asyncio.coroutine
+def aiohttp_multiple_methods_get(name, **kwargs):
+    data = {'method': 'get'}
+    return ConnexionResponse(
+        body=data
+    )
+
+
+@asyncio.coroutine
+def aiohttp_multiple_methods_post(name, **kwargs):
```

Comment:
The tests with payload should also check the payload. Instead of passing a name as a path variable better to pass it as a json payload and extract it here.
I'm trying to debug it, because it doesn't work for my [demo app](https://github.com/prawn-cake/tryConnexion).
```
@asyncio.coroutine
def test_connexion(test_client):
    app = connexion.AioHttpApp(__name__)
    app.add_api('swagger.yaml', base_path='/api')
    aiohttp_app = app.app
    client = yield from test_client(aiohttp_app)

    routes = aiohttp_app.router.routes()

    resp = yield from client.get('/1')
    assert resp.status == 404

    resp = yield from client.get('/api/pets')
    json_data = yield from resp.json()
    assert json_data == []

    pet = {"id": 1, "name": "Tosi", "animal_type": "cat"}
    resp = yield from client.put('/api/pets/1', json=pet)
```

It returns some weird errors like:
```
connexion/decorators/validation.py", line 109, in wrapper
if data is None and len(request.body) > 0 and not self.is_null_value_valid:
TypeError: object of type 'FlowControlStreamReader' has no len()
```

Then I discovered that this functionality is not tested in your PR. Please, check it as well.
If I manage to debug it, I will send a PR to your branch

Reply:
OK. So the problem is request body is a stream and due to its' async nature you have to await for it and it should happen when you create `ConnexionRequest`. 

Currently if fails during request body validation.
The hard part is that the validation process is hapenning in the middleware which is a coroutine for aiohttp api which make validation code incompatible.

I'm still investigating into it. Just wanna give you outline of the problem

Reply:
@prawn-cake yes, I saw this problem too. I know how to fix it. In the response validation happens the same situation and I solve this with extra module import. I thing I can do something like this on body validation too.

Reply:
@dutradda 
I sent a [PR](https://github.com/dutradda/connexion/pull/1) to your branch. It doesn't fix the problem here, but I cleaned up some things a little bit + fixed some aiohttp deprecated methods (new `middleware` style + `request.can_read_body` calls).

Cool, if you're familiar with the problem, let's try to fix it then.